internal/ingest/ingest.go

```diff
@@ -225,6 +218,45 @@ func (ing *Ingester) Sync(ctx context.Context, peerID peer.ID, peerAddr multiadd
 	return out, nil
 }
 
+func (ing *Ingester) makeLimitedDepthSelector(peerID peer.ID, depth int64, nolatest bool) (ipld.Node, error) {
+	// Consider the value of -1 as no-limit, similar to config.Ingest.
+	var rLimit selector.RecursionLimit
+	if depth == -1 {
+		rLimit = selector.RecursionLimitNone()
+	} else if depth == 0 {
+		rLimit = ing.cfg.AdvertisementRecursionLimit()
+		// Override the value of depth with config.Ingest value for logging purposes.
+		depth = ing.cfg.AdvertisementDepthLimit
+	} else {
+		rLimit = selector.RecursionLimitDepth(depth)
+	}
+	log := log.With("depth", depth)
+
+	var stopAt ipld.Link
+	if !nolatest {
+		latest, err := ing.GetLatestSync(peerID)
+		if err != nil {
+			return nil, err
+		}
+
+		if latest != cid.Undef {
+			stopAt = cidlink.Link{Cid: latest}
+		}
+	}
+	// The stop link may be nil, in which case it is treated as no stop link.
+	// Log it regardless for debugging purposes.
+	log = log.With("stopAt", stopAt)
```

Comment:
I think this should be moved into `if latest != cid.Undef {`, so that it is not logged if not present.

Reply:
I think we want to log it regardless for debugging purposes. Now that this function is refactored, the log including this is only made in debug level, and if debug level is set we probably want to know whether stop node was nil or not.